### PR TITLE
arm64: dts: qcom: msm8916-samsung-j{3,5x}: Add NFC

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**
Also add `pvdd-supply` support to `nxp-nci`.

cc @marohinmark

### CODE REVIEW

File: drivers/nfc/nxp-nci/i2c.c [hunk 1]
```diff
@@ -297,6 +298,18 @@ static int nxp_nci_i2c_probe(struct i2c_client *client,
 	if (r < 0)
 		return r;
 
+	phy->supplies[0].supply = "pvdd";
+	r = devm_regulator_bulk_get(dev, ARRAY_SIZE(phy->supplies),
+				      phy->supplies);
```
Comment (stephan-gh) [inline drivers/nfc/nxp-nci/i2c.c:303]: It's a bit weird to use regulator bulk for a single regulator. Maybe use the single regulator APIs instead (`devm_regulator_get`, `regulator_enable` etc).

File: drivers/nfc/nxp-nci/i2c.c [hunk 2]
```diff
@@ -297,6 +298,18 @@ static int nxp_nci_i2c_probe(struct i2c_client *client,
...
+	if (r < 0)
+		return dev_err_probe(dev, r, "Failed to get regulators\n");
+
+	r = regulator_bulk_enable(ARRAY_SIZE(phy->supplies), phy->supplies);
```
Comment (stephan-gh) [inline drivers/nfc/nxp-nci/i2c.c:307]: You need to disable the regulator again somewhere when the driver is unloaded. You can use `devm_action_or_reset()` similar to https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=1d2e91a2db664fa5c0b935fe45314759f6f1fdc5

File: Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml [hunk 3]
```diff
@@ -24,6 +24,9 @@ properties:
   firmware-gpios:
     description: Output GPIO pin used to enter firmware download mode
 
+  vddio-supply:
+    description: Optional regulator used for some PN547 controller
```
Comment (wonderfulShrineMaidenOfParadise) [inline Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml:28]:
```suggestion
  pvdd-supply:
    description: Optional regulator used for some PN544 and PN547 controller
```
- Typo. should be `pvdd-supply`
- Actually it's "pad supply"

File: drivers/nfc/nxp-nci/i2c.c (hunk 1 above)
Comment (wonderfulShrineMaidenOfParadise) [reply drivers/nfc/nxp-nci/i2c.c]: Copied single regulator APIs from the other drivers.

File: drivers/nfc/nxp-nci/i2c.c (hunk 2 above)
Comment (wonderfulShrineMaidenOfParadise) [reply drivers/nfc/nxp-nci/i2c.c]: Put a `regulator_disable(phy->pvdd);`  in `nxp_nci_i2c_remove()`.

Comment (stephan-gh) [reply drivers/nfc/nxp-nci/i2c.c]: The problem with that is that you need to also place `regulator_disable(phy->pvdd);` in all error paths of the probe() function. Consider that the `request_threaded_irq()` after your `regulator_enable()` fails. Then you are expected to undo all effects made in the probe() function.

This means you'd need to write:

```c
	r = regulator_enable(phy->pvdd);
	if (r < 0) {
		nfc_err(&client->dev,
			"Failed to enable regulator pvdd: %d\n", r);
		return r;
	}

	r = request_threaded_irq(client->irq, NULL,
				 nxp_nci_i2c_irq_thread_fn,
				 IRQF_TRIGGER_RISING | IRQF_ONESHOT,
				 NXP_NCI_I2C_DRIVER_NAME, phy);
	if (r < 0) {
		nfc_err(&client->dev, "Unable to register IRQ handler\n");
		regulator_disable(phy->pvdd); // <-----
		return r;
	}

	return 0;
```

The `devm_action_or_reset()` does that for you automatically when probe() returns an error code.

**Pull request closed without merging**